### ISSUE TITLE
Overwriting a getter from a mixin calls the original getter

### ISSUE THREAD

**Issue by boris-petrov** [emberjs/ember.js#18829]
Ember 3.16.3.

```javascript
import EmberObject from '@ember/object';
import Mixin from '@ember/object/mixin';

class Model extends EmberObject {
  get bar() {
    console.log('bar');
  }
}
Object.defineProperty(Model.prototype, 'foo', {
  get() {
    console.log('foo');
  },
});

const mixin = Mixin.create({
  get foo() {
  },
  get bar() {
  },
});
const NewModel = Model.extend(mixin);

NewModel.create();
```

Both `foo` and `bar` are printed in the console when this code executes. They both shouldn't.

cc @pzuraq

**Comment by lifeart** [comment]
I think issue from https://github.com/emberjs/ember.js/blob/f73d8440d19cf86a10c61ddb89d45881acfcf974/packages/%40ember/-internals/metal/lib/mixin.ts#L190

**Comment by boris-petrov** [comment]
I believe this issue has been resolved (I believe as of Ember 4.12). Thanks!
